src/test/java/com/btto/core/service/AccessServiceImplTest.java

```diff
@@ -0,0 +1,804 @@
+
+        assertFalse(accessService.hasUserRight(admin, 2, AccessService.UserRight.REMOVE));
+    }
+
+    @Test
+    void testThatAdminCanRemoveUserWithTheSameCompany() {
+        final Company company = new MockCompanyBuilder(1).build();
+        final User admin = new MockUserBuilder(1).role(Role.Admin).company(company).build();
+        final User user = new MockUserBuilder(2).company(company).build();
+
+        when(userService.find(user.getId())).thenReturn(Optional.of(user));
+
+        assertTrue(accessService.hasUserRight(admin, user.getId(), AccessService.UserRight.REMOVE));
+    }
+
+    @Test
+    void testThatAdminCantRemoveUserWithTheDifferentCompany() {
+        final Company userCompany = new MockCompanyBuilder(1).build();
+        final Company adminCompany = new MockCompanyBuilder(2).build();
+        final User admin = new MockUserBuilder(1).role(Role.Admin).company(adminCompany).build();
+        final User user = new MockUserBuilder(2).company(userCompany).build();
+
+        when(userService.find(user.getId())).thenReturn(Optional.of(user));
+
+        assertFalse(accessService.hasUserRight(admin, user.getId(), AccessService.UserRight.EDIT));
+    }
+
+    @Test
+    void testThanUserCantCreateUser() {
+        final Company company = new MockCompanyBuilder(1).build();
+        final User user = new MockUserBuilder(1).company(company).build();
+
+        assertFalse(accessService.hasUserRight(user, null, AccessService.UserRight.CREATE));
+    }
+
+    @Test
+    void testThanAdminCanCreateUser() {
+        final Company company = new MockCompanyBuilder(1).build();
+        final User admin = new MockUserBuilder(1).role(Role.Admin).company(company).build();
+
+        assertTrue(accessService.hasUserRight(admin, null, AccessService.UserRight.CREATE));
+    }
+
+    @Test
+    void testThanAdminWithoutCompanyCantCreateUser() {
+        final User admin = new MockUserBuilder(1).role(Role.Admin).build();
+
+        assertFalse(accessService.hasUserRight(admin, null, AccessService.UserRight.CREATE));
+    }
+
+    @Test
+    void testThanUserCanSetStatus() {
+        final Company company = new MockCompanyBuilder(1).build();
+        final User user = new MockUserBuilder(1).company(company).build();
+
+        assertTrue(accessService.hasUserRight(user, user.getId(), AccessService.UserRight.SET_STATUS));
+    }
+
+    @Test
+    void testThanAdminCantSetStatusToAnotherUser() {
+        final Company company = new MockCompanyBuilder(1).build();
+        final User admin = new MockUserBuilder(1).company(company).role(Role.Admin).build();
+        final User user = new MockUserBuilder(2).company(company).build();
+
+        assertFalse(accessService.hasUserRight(admin, user.getId(), AccessService.UserRight.SET_STATUS));
+    }
+
+    @ParameterizedTest
+    @EnumSource(AccessService.DepartmentRight.class)
+    void testThatAdminWithoutCompanyCanDoNothingWithDepartments(final AccessService.DepartmentRight right) {
```

Comment:
I think correctly CantDoAnything, because before you use 'cant' when expected false. My opinion, maybe I wrong.

Reply:
`can do nothing` means the same that `Can't do anything` - but in general you're right, let's use explicit, clear ways